src/main/java/com/flightstats/hub/dao/aws/ClusterContentService.java

```diff
@@ -143,33 +149,66 @@ public ClusterContentService(
     }
 
     @Override
-    public ContentKey insert(String channelName, Content content) throws Exception {
+    public ContentKey insert(String channelName, Content content) {
+        Content spokeContent = calculateIndexIfLarge(channelName, content);
+
+        SortedSet<ContentKey> keys = setStableCache(channelName,
+                () -> new TreeSet<>(
+                        Arrays.asList(spokeWriteContentDao.insert(channelName, spokeContent))));
+
+        if (isWriteable(channelName)) {
+            s3SingleWrite(channelName, keys.first());
+        }
+        return keys.first();
+    }
+
+    private Content calculateIndexIfLarge(String channelName, Content content) {
         Content spokeContent = content;
         if (content.isLarge()) {
             s3LargePayloadContentDao.insert(channelName, content);
             spokeContent = largeContentUtils.createIndex(content);
         }
-        final ContentKey key = spokeWriteContentDao.insert(channelName, spokeContent);
-        if (isWriteable(channelName)) {
-            s3SingleWrite(channelName, key);
+        return spokeContent;
+    }
+
+    private SortedSet<ContentKey> setStableCache(String channelName, Supplier<SortedSet<ContentKey>> supplier) {
+        // after stable() seconds, we want to write the cache entry if the write succeeded
+        AtomicReference<ContentKey> ref = new AtomicReference<>();
+
+        ScheduledFuture future = zkCacheStateUpdateExecutor.schedule(() -> {
```

Comment:
Yep.
It's an unbounded queue, so there shouldn't be a hard limit.  4 is the core size, but the implementation will scale up to MAX_INT threads as needed to unload the queue.  Overall this will reduce load from hub itself in /latest->*/internal/latest for spoke-resident latest items at the cost of two ZK ops per insert.  
We could try to coalesce events for channels like FAA feeds in iad.prod, but we'd have to play looser with the meaning of "stable" time.
TBH, I would love to change hub's concept of "stable" to "cluster acknowledged" or "majority acknowledged".  It would simplify code like this and the reads that self-select "stable" time limits, and, it'd be more robust.  That's a decent chunk of work though.

Reply:
I didn't think about what we gained from this change beyond getting rid of that scheduled task and the wonky ZK update side effect. That doesn't seem so bad.

I have a feeling that if/when we replace the storage implementation for the hub with something third party, we'll be able to stop worrying about stable time. Not knowing the timeline for that feels like punting on deciding whether we should put aside some time or a task to scope that work out.